Data/Vector/Generic.hs

```diff
@@ -1334,6 +1334,14 @@ filterM :: (Monad m, Vector v a) => (a -> m Bool) -> v a -> m (v a)
 {-# INLINE filterM #-}
 filterM f = unstreamM . Bundle.filterM f . stream
 
+mapMaybeM :: (Monad m, Vector v a, Vector v b) => (a -> m (Maybe b)) -> v a -> m (v b)
+{-# INLINE mapMaybeM #-}
+mapMaybeM f = unstreamM . Bundle.mapMaybeM f . stream
+
+imapMaybeM :: (Monad m, Vector v a, Vector v b)
+      => (Int -> a -> m (Maybe b)) -> v a -> m (v b)
+imapMaybeM f = unstreamM . Bundle.mapMaybeM (\(i, a) -> f i a) . Bundle.indexed . stream
```

Comment:
Other indexed functions in this module seem to use `uncurry f` instead of `(\(i, a) -> f i a)`—perhaps it would be worth sticking to this convention? (I'm not sure if using one or the other has any actual ramifications on strictness properties.)

Reply:
I never use `uncurry` because it introduces laziness I virtually never need. I bet the compiler often fixes that, but I see no reason to tempt fate.

Reply:
i wonder if the `uncurry` thing is actually necessary in general. i tend to prefer not using it as well.